lib/polo/collector.rb

```diff
@@ -14,15 +14,25 @@ def initialize(base_class, id, dependency_tree={})
     # ActiveSupport::Notifications block and collecting every generate SQL query.
     #
     def collect
-      unprepared_statement do
-        ActiveSupport::Notifications.subscribed(collector, 'sql.active_record') do
-          base_finder = @base_class.includes(@dependency_tree).where(@base_class.primary_key => @id)
-          collect_sql(@base_class, base_finder.to_sql)
-          base_finder.to_a
+      base_finder = @base_class.includes(@dependency_tree).where(@base_class.primary_key => @id)
+      enumerable = Enumerator.new do |yielder|
+        collect_sql(@base_class, base_finder.to_sql)
+        unprepared_statement do
+          ActiveSupport::Notifications.subscribed(collector, 'sql.active_record') do
+            base_finder.find_in_batches.with_index do |batch, batch_index|
```

Comment:
A side effect to consider is that `find_in_batches` will ignore any `order` set on the passed in scope. I can't imagine why someone would care about the order of their scrubbed records, but technically I suppose it is a breaking change.